### PR TITLE
Add color issues section

### LINKED ISSUE #143: Read info about known color issues
## Scope

- add an unobtrusive information section that is expanded on larger screens and collapsed on smaller screen to help interested people understand the known color issues
- this section can go into as much detail as needed and even include color swatches to help people understand the foreground color issue related to alpha

## Mockups

Here are the warning-related Figma file links.
Note: I don't know why Figma is suddenly displaying green lines around various bits of the mockups, and I haven't found the way to turn that view off yet.

**Background Alpha Alert + Color Issue Info**
https://www.figma.com/file/JtQTGIpwoaOcqcWfsoeOk1/Browser-bugs-%26-other-warnings?type=design&node-id=112-2&mode=design&t=lOyosYvXrteuMN5Z-0

**Foreground Alpha Alert**
https://www.figma.com/file/JtQTGIpwoaOcqcWfsoeOk1/Browser-bugs-%26-other-warnings?type=design&node-id=12%3A348&mode=design&t=4CwzTDrU2fakSiQ4-1

**Mobile Alerts + Color Issue Info**
https://www.figma.com/file/JtQTGIpwoaOcqcWfsoeOk1/Browser-bugs-%26-other-warnings?type=design&node-id=83%3A10&mode=design&t=4CwzTDrU2fakSiQ4-1

**Mobile Alerts + Color Issue Info 2**
https://www.figma.com/file/JtQTGIpwoaOcqcWfsoeOk1/Browser-bugs-%26-other-warnings?type=design&node-id=83%3A11&mode=design&t=4CwzTDrU2fakSiQ4-1

### PR DESCRIPTION
**Pull request by jamesnw**
## Description
Adds a section with space for more info on known color issues. Currently, only the Gamut mapping issue is documented- the Alpha caveats should be added with the related PRs.

Fixes #143 

## Steps to test/reproduce

1. Open the page on a wide screen, see that the section defaults to open.
2. Open the page on a narrow screen, see that the section defaults to closed.
3. Open/close is not responsive, so changing the viewport won't change the state
4. When the contrast doesn't meet AA Normal, it uses the same low contrast colors as the ratios.

## Show me
image
image

### CODE REVIEW

**Comment by jamesnw** [comment]
This will need some design help from @stacyk or @dvdherron - there was some conflicts with the other definition list in use. I added the styles to the component, but assume they should be located elsewhere.

File: src/lib/components/ratio/ColorIssues.svelte
Comment (jgerigmeyer) [inline src/lib/components/ratio/ColorIssues.svelte:6]: @jamesnw Does this change (in https://github.com/oddbird/oddcontrast/pull/146/commits/af6b86db0f0c734ae876a45aabfbaf12b59c3e43) make sense? I figure this should probably use the same `pass` logic (and might as well use the same reactive boolean) as the ratio display.

Comment (jamesnw) [reply src/lib/components/ratio/ColorIssues.svelte:6]: I'm not opposed to reuse, but I wanted to use the AA Normal contrast ratio, where `pass` uses AA Large- I think Normal makes sense, given the text size.

File: src/lib/components/ratio/ColorIssues.svelte
Comment (dvdherron) [inline src/lib/components/ratio/ColorIssues.svelte:34]: @stacyk This spacing was a little tricky. When the colors are passing this section seems sort of misaligned. 

image

But that padding is needed once the colors fail and the background is added

image

File: src/lib/components/ratio/ColorIssues.svelte
Comment (dvdherron) [inline src/lib/components/ratio/ColorIssues.svelte:39]: @stacyk this grid (and the paragraph spacing) takes into account additional [issues being added later](https://github.com/oddbird/oddcontrast/pull/145#issuecomment-1854366303).

**Comment by dvdherron** [comment]
@stacyk This is ready for style review.

File: src/sass/patterns/_lists.scss
Comment (stacyk) [inline src/sass/patterns/_lists.scss:29]: @dvdherron On the dl below, I wonder if we will want to add something like dl:last-child and give it a zero margin-block-end. Otherwise the spacing gets a little large between sections since the gap is already adding space to the grid rows.

File: src/lib/components/ratio/index.svelte
Comment (stacyk) [inline src/lib/components/ratio/index.svelte:79]: @dvdherron when at the smallest size, I wonder if we want to adjust or remove the gap on [data-layout='results']  since it is adding a double gap between the pass/fail section and the known issues section even though we have the contrast-defined section set to display none. 

Screenshot 2023-12-18 at 10 03 21 AM

**Review by stacyk — COMMENTED** [review]
I think we will want to add link colors to data-pass=false

something like:

[data-pass=false] {
--link: var(--status-result-fg);
--link-focus: var(--status-result-fg);
}

I know I didn't suggest changing the color on hover/focus but the underline grows when you hover, and when you focus the underline grows and there is a box.

File: src/lib/components/ratio/ColorIssues.svelte
Comment (dvdherron) [reply src/lib/components/ratio/ColorIssues.svelte]: @stacyk This worked to get the text aligned correctly, but that made the marker end up in a weird place. I took most of your suggestions above but also added a custom marker to the other side of the summary text:

image

image

What do you (and @Sondra) think of this placement?

File: src/lib/components/ratio/index.svelte
Comment (dvdherron) [reply src/lib/components/ratio/index.svelte]: I made this smaller in https://github.com/oddbird/oddcontrast/pull/146/commits/72e099cb82200f7bf98680625eb19eb26f6c08bf

image

File: src/lib/components/ratio/ColorIssues.svelte
Comment (stacyk) [inline src/lib/components/ratio/ColorIssues.svelte:80]:
```suggestion
      transform: rotate(0);
      transition: transform 150ms;
```

File: src/lib/components/ratio/ColorIssues.svelte
Comment (stacyk) [inline src/lib/components/ratio/ColorIssues.svelte:71]: Does this project not require spaces between these blocks?

File: src/lib/components/ratio/ColorIssues.svelte
Comment (stacyk) [reply src/lib/components/ratio/ColorIssues.svelte:59]: I'm going to open a new issue for consideration and approve this PR as is now.

**Review by stacyk — APPROVED** [review]

**Comment by stacyk** [comment]
@jgerigmeyer I am assigning this to you because I see you left comments but haven't approved this PR yet.

**Review by stacyk — CHANGES_REQUESTED** [review]
I thought this was an issue that was fixed, but I see that Safari adds their own marker, so what I thought was just a thing in progress was actually a browser difference.  
Screenshot 2023-12-21 at 3 19 25 PM

I'll see what we can do since I use Safari as a primary, might be easier for me to test.

File: src/sass/patterns/_disclosure.scss
Comment (stacyk) [inline src/sass/patterns/_disclosure.scss:40]: @mirisuzanne @dvdherron Is there a bug in Safari that doesn't allow this before pseudo element to transition when we add the [open] attribute? I am able to see the transitions if I just toggle the properties (see screen recording) but not when I click to open/close it. I made it 1s for testing purposes so it is obvious. 




https://github.com/oddbird/oddcontrast/assets/1581694/7eaba154-ae60-4728-b9a3-b1a81206415d

**Review by jgerigmeyer — APPROVED** [review]
LGTM 🚀 

> @mirisuzanne @dvdherron Is there a bug in Safari that doesn't allow this before pseudo element to transition when we add the [open] attribute? I am able to see the transitions if I just toggle the properties (see screen recording) but not when I click to open/close it.

@stacyk Is this issue blocking merge for this PR? Or something we want to capture in a new story?

**Comment by stacyk** [comment]
We shouldn't block the pr because of the transition, but i think there was a ridiculous duration and delay on that transition for testing that would need to be adjusted.

File: src/lib/components/ratio/ColorIssues.svelte
Comment (dvdherron) [reply src/lib/components/ratio/ColorIssues.svelte]: That make sense and I think it looks okay with the indentation on small-medium screens. Fixed in https://github.com/oddbird/oddcontrast/pull/146/commits/137bc34bb979f43ec75d80bdffae557a912ac34b

**Passing**

wide:

image

medium:

image

small:

image

**Failing**

wide: 

image

medium:

image

small:

image

File: src/sass/patterns/_disclosure.scss
Comment (dvdherron) [reply src/sass/patterns/_disclosure.scss]: I tried searching for a bug related to this but haven't been able to find anything just yet. I know it shouldn't be needed but does adding a vendor prefix change anything @stacyk ?

**Comment by dvdherron** [comment]
@stacyk I think my latest changes address everything except for the possible Safari transition bug. If you want to give this a last look I think this is soon ready to merge.

File: src/sass/patterns/_disclosure.scss
Comment (stacyk) [reply src/sass/patterns/_disclosure.scss]: I just tried the vendor prefix and I couldn't find anything either. I found this: https://bugs.webkit.org/show_bug.cgi?id=213349 

So it looks like it was a bug, then it was fixed, and then it came back in version 16. I don't know if it is helpful to send them our use case once this is merged?

**Review by stacyk — APPROVED** [review]
yay! I love this app. Nice work everyone!

**Pull request merged**
